config/redis.js

```diff
@@ -3,25 +3,49 @@
 const config = require('config');
 
 const queueConfig = config.get('queue');
-const redisConfig = queueConfig.redisConnection;
+const connectionType = queueConfig.connectionType;
+
+if (!connectionType && (connectionType !== 'redis' || connectionType !== 'redisCluster')) {
+    throw new Error(
+        `'${connectionType}' is not supported in connectionType, 'redis' or 'redisCluster' can be set for the queue.connectionType setting`
+    );
+}
+
+const redisConfig = queueConfig[`${connectionType}Connection`];
 const connectionDetails = {
-    pkg: 'ioredis',
-    host: redisConfig.host,
-    options: {
+    redisOptions: {
         password: redisConfig.options && redisConfig.options.password,
         tls: redisConfig.options ? redisConfig.options.tls : false
-    },
-    port: redisConfig.port,
-    database: redisConfig.database
+    }
 };
 
+let queueNamespace;
+
+// for redisCluster config
+if (connectionType === 'redisCluster') {
+    connectionDetails.redisClusterHosts = redisConfig.hosts;
+    connectionDetails.slotsRefreshTimeout = parseInt(redisConfig.slotsRefreshTimeout, 10);
+    // NOTE: node-resque has an issue  in multi-key operation for Redis Cluster
+    // https://github.com/actionhero/node-resque/issues/786
+    // so we have to set the namespace option with a hash tag so that the resque's keys are set in the same slots in Redis Cluster
+    // https://redis.io/docs/manual/scaling/#redis-cluster-data-sharding
+    queueNamespace = 'resque:{screwdriver-resque}';
+} else {
+    // for non-cluster redis config
+    connectionDetails.redisOptions.host = redisConfig.host;
+    connectionDetails.redisOptions.port = redisConfig.port;
+    connectionDetails.redisOptions.database = redisConfig.database;
```

Comment:
Do you need to add `connectionDetails.redisOptions.pkg` ?

Reply:
`pkg` option is no longer needed, because the option was passed to `Resque` constructor to instantiate the redis object like [this](https://github.com/screwdriver-cd/queue-service/pull/57/files#diff-afa7a0a70da4b0fa42349adcdb862c6f079f980b4051ca651d929290f34a29c7L47) before, but this PR changed it to pass a redis instance to `Resque` constructor to support `Redis.Cluster` instead of pkg option. It seems we can't use pkg option to instantiate Redis.Cluster.

I reffered to the official documentation.
>You can also pass redis client directly.
https://github.com/actionhero/node-resque#configuration-options

Reply:
Since the redis instance is generated from the ioredis package, it certainly seems unnecessary to specify pkg. Thanks.